templates/frontend/objects/article_summary.tpl

```diff
@@ -60,6 +61,12 @@
 	{if $hasAccess}
 		<ul class="galleys_links">
 			{foreach from=$article->getGalleys() item=galley}
+				{if $primaryGenreIds}
+					{assign var="file" value=$galley->getFile()}
+					{if !$file || !in_array($file->getGenreId(), $primaryGenreIds)}
+						{php}continue;{/php}
+					{/if}
+				{/if}
```

Comment:
What if an article only has supplementary files i.e. there is no$primaryGenreIds -- should the galley_link.tpl below be included however or shall the if statement enclose that too?